### ISSUE TITLE
Support for hyperlinks 

### ISSUE THREAD

**Issue by sarauniverse** [johngray1965/PdfiumAndroidKt#25]
It will be very useful if the hyperlink annotations could be rendered and they are clickable.

**Comment by johngray1965** [comment]
I just put out a version that should handle the annotations.    There's callback that gets called when they are clicked, but there's no way to access at the moment.   We'll look into making the callback accessible (callback from the native code to the Java/Kotlin code is a little tricky).

**Comment by bvitaliyg** [comment]
From what I understand, handling the links could be very tricky. The library renders everything in a Bitmap. It can also provide you a list of links with their coordinates and references. But creating the overlay that handles the links seems to be a task more suitable for PDF Viewer, not renderer.

The links click will depend on the platform (Views? Compose?), render scale (as the hitboxes needs to be scaled as well), rendered image position on the screen, etc. Since this library seems to be a Kotlin wrapper around Pdfium, having built-in Viewer seems a bit overkill. And if the Viewer is going to be implemented, it needs to support a lot of stuff – text selections, zoom, links, etc. The current API already provides all the data to have this implemented, but there's no Viewer (yet). Maybe it could be done as another library that uses this under the hood to get all the information, or another module in this library (e.g. `viewer-compose`). But so far this looks like an overkill task IMHO if the goal of this library is just to have Pdfium methods bindings in Kotlin.

**Comment by johngray1965** [comment]
My plan is just to expose the api's so the user has the information on what the links are there, and where they are.  We don't do UI here.   We just give the app developer the tools.  There are just a few API to expose.  And probably wrap a new type.  It's not a big deal.  So really, as you said, we do kotlin bindings.  So I'm just adding some new kotlin bindings.

**Comment by johngray1965** [comment]
in fact, I plan to do kotlin bindings for:
```
FPDF_EXPORT FPDF_PAGELINK FPDF_CALLCONV
FPDFLink_LoadWebLinks(FPDF_TEXTPAGE text_page);

FPDF_EXPORT int FPDF_CALLCONV FPDFLink_CountWebLinks(FPDF_PAGELINK link_page);

FPDF_EXPORT int FPDF_CALLCONV FPDFLink_GetURL(FPDF_PAGELINK link_page,
                                              int link_index,
                                              unsigned short* buffer,
                                              int buflen);

```

I'll have to wrap FPDF_PAGELINK.  That's it.    An hour or two.   Might take longer to write up something explaining to the user who wanted it what they are getting and how to use it.

**Comment by johngray1965** [comment]
Hello, I've added Android Kotlin bindings for the APIs in Pdfium for getting the links on the page and where they are.  When we gave you a rendered page, its a bitmap, which doesn't support links.  And this project doesn't do UI, just kotlin binding for pdfium.   As Vitaliy pointed out, the UI is complex matter.   Made more complex by the fact there are now two UI systems for Android layouts and Compose.    

While, I have written all the code to talk the APIs, it has not been tested yet.  @sarauniverse
